tools/nnyeah/nnyeah/ConstructorTransforms.cs

```diff
@@ -48,15 +54,38 @@ static bool IsIntPtrCtor (MethodDefinition d)
             return isSingle || isDouble;
         }
 
-        static bool IsNSObjectDerived (TypeReference? typeReference)
+        bool IsNSObjectDerived (TypeReference? typeReference)
         {
-            if (typeReference is null) {
+            if (typeReference is null)
                 return false;
+            var initialTypeReference = typeReference!;
```

Comment:
The suppressing is not needed, ! can yo away.
```suggestion
            var initialTypeReference = typeReference;
```